satellite/satellitedb/usercredits.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/satellite/console"
+	dbx "storj.io/storj/satellite/satellitedb/dbx"
+)
+
+type usercredits struct {
+	db *dbx.DB
+}
+
+type updateInfo struct {
+	id      int
+	credits int
+}
+
+// TotalReferredCount returns the total amount of referral a user has made based on user id
+func (c *usercredits) TotalReferredCount(ctx context.Context, id uuid.UUID) (int64, error) {
+	totalReferred, err := c.db.Count_UserCredit_By_ReferredBy(ctx, dbx.UserCredit_ReferredBy(id[:]))
+	if err != nil {
+		return totalReferred, errs.Wrap(err)
+	}
+
+	return totalReferred, nil
+}
+
+// GetAvailableCredits returns all records of user credit that are not expired or used
+func (c *usercredits) GetAvailableCredits(ctx context.Context, referrerID uuid.UUID, expirationEndDate time.Time) ([]console.UserCredit, error) {
+	availableCredits, err := c.db.All_UserCredit_By_UserId_And_ExpiresAt_Greater_And_CreditsUsedInCents_Less_CreditsEarnedInCents_OrderBy_Asc_ExpiresAt(ctx,
+		dbx.UserCredit_UserId(referrerID[:]),
+		dbx.UserCredit_ExpiresAt(expirationEndDate),
+	)
+	if err != nil {
+		return nil, errs.Wrap(err)
+	}
+
+	return fromDBX(availableCredits)
```

Comment:
Let's use a better name than `fromDBX` this package contains other implementations. e.g. `userCreditsFromDBX`